config/builtin_locks.go

```diff
@@ -0,0 +1,26 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/gruntwork-io/terragrunt/locks"
+	"github.com/gruntwork-io/terragrunt/locks/dynamodb"
+)
+
+// lockFactory provides an implementation of Lock with the provided configuration map
+type lockFactory func(map[string]string) (locks.Lock, error)
+
+// lookupLock returns the implementation for the named lock or returns an error if
+// it is not found
+func lookupLock(name string, conf map[string]string) (locks.Lock, error) {
+	f, ok := builtinLocks[name]
```

Comment:
I generally prefer more descriptive variable names than `f` or `ok`. Perhaps `factory` and `containsFactory`?